lib/matplotlib/figure.py

```diff
@@ -1936,11 +1830,11 @@ def subplot_mosaic(self, mosaic, *, sharex=False, sharey=False,
         # special-case string input
         if isinstance(mosaic, str):
             mosaic = self._normalize_grid_string(mosaic)
-            per_subplot_kw = {
-                tuple(k): v for k, v in per_subplot_kw.items()
+            per_subthing_kw = {
+                tuple(k): v for k, v in per_subthing_kw.items()
             }
 
-        per_subplot_kw = self._norm_per_subplot_kw(per_subplot_kw)
+        per_subthing_kw = self._norm_per_subplot_kw(per_subthing_kw)
```

Comment:
Does this normalization also need to be changed?

Reply:
My first thought was `_flatten_subthing_kw`  but your version sounds more descriptive of what the method's actually being used for. Should I go ahead and change it to that?

Reply:
`flatten` is also pretty good (maybe better).  In general, the prior should be "Tom is bad at naming things".

Reply:
Is something like ` _check_duplication_and_flatten_subthing_kw ` too long?

Reply:
it is an internal method, so no ;)

Reply:
Got you :) I'll change it to that if you're ok with it, then.